sls-versions/src/main/java/com/palantir/sls/versions/CompactVersion.java

```diff
@@ -112,7 +112,7 @@ private static long encodePriority2(SlsVersionType type) {
         throw new SafeIllegalArgumentException("Unknown SlsVersionType", SafeArg.of("slsVersionType", type));
     }
 
-    private static long encode20b(int value, @CompileTimeConstant String component) {
+    private static long encode20b(int value, @CompileTimeConstant final String component) {
```

Comment:
Would prefer to omit the `final` here if possible?

Reply:
done - I hadn't read carefully, thought this was a compiler error but it was only a warn 👍